docs/csharp/fundamentals/program-structure/main-command-line.md

```diff
@@ -105,64 +108,80 @@ Execution succeeded
 Return value = 0
 ```
 
-### Async Main return values
+### Main return Task
+
+When the application entry point returns a `Task` or `Task<int>`, the compiler generates a new entry point that calls the entry point method declared in the application code. Assuming that this entry point is called `$GeneratedMain`, the compiler generates the following code for these entry points:
```

Comment:
I'm not sure it makes sense to document all this detail. It's an implementation detail. The compiler does plenty of codegen for you. This feels similar.

Reply:
I agree. The salient detail is that the process won't exit until the task returned from Main completes. How that happens is implementation detail.

Reply:
I agree, should we delete this part?

Reply:
I agree as well. I'd rather state that the program doesn't exit until the task completes (either successfully or faulted)

Reply:
I rewrote it to explicitly notice that the returned task is waited, without the part concerning the boilerplate.

Reply:
This modification has more implication I anticipated, so I prefer deal this in dedicated PR.
I created the issue #41457 to follow up it.